lib/api/src/utils/mem/mod.rs

```diff
@@ -103,8 +106,11 @@ impl<'a, T: ValueType> WasmRef<'a, T> {
     #[inline]
     pub fn read(self) -> Result<T, MemoryAccessError> {
         let mut out = MaybeUninit::uninit();
-        let buf =
-            unsafe { slice::from_raw_parts_mut(out.as_mut_ptr() as *mut u8, mem::size_of::<T>()) };
+        let buf_ptr = out.as_mut_ptr() as *mut u8;
+        if !buf_ptr.is_aligned() {
+            return Err(MemoryAccessError::UnalignedPointer);
+        }
+        let buf = unsafe { slice::from_raw_parts_mut(buf_ptr, mem::size_of::<T>()) };
```

Comment:
Isn't `.is_aligned()` always going to return true due to [this](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#layout-1)?

> _MaybeUninit<T> is guaranteed to have the same size, alignment, and ABI as T:_

Reply:
Depends... These guarantees only hold as long as we don't have unsafe code, and we do have a lot of unsafe code near the syscall boundaries. Still, @xdoardo would know better.

Reply:
But also, since we're casting to `*mut u8`... Let's add a test?

Reply:
[b7cd7a4](https://github.com/wasmerio/wasmer/pull/5449/commits/b7cd7a4a384df9e95c8137924ba58412caa5e381)